v2/tools/generator/internal/functions/property_assignment_function_builder.go

```diff
@@ -369,10 +370,87 @@ func (*PropertyAssignmentFunctionBuilder) selectIdenticallyNamedProperties(
 	sourceProperties conversions.ReadableConversionEndpointSet,
 	destinationProperties conversions.WritableConversionEndpointSet,
 	assign func(reader *conversions.ReadableConversionEndpoint, writer *conversions.WritableConversionEndpoint) error,
-	_ *conversions.PropertyConversionContext,
+	ctx *conversions.PropertyConversionContext,
 ) error {
 	for destinationName, destinationEndpoint := range destinationProperties {
 		if sourceEndpoint, ok := sourceProperties[destinationName]; ok {
+			// We have properties with the same name, we also need them to have the same path.
+			// This effectively means they were flattened from the same original property.
+			// If they don't, we can't match them up.
+			if sourceEndpoint.Endpoint().Path() != destinationEndpoint.Endpoint().Path() {
+				continue
+			}
+
+			err := assign(sourceEndpoint, destinationEndpoint)
+			if err != nil {
+				return errors.Wrapf(err, "assigning %s", destinationName)
+			}
+		}
+	}
+
+	return nil
+}
+
+// selectIdenticallyPathedProperties matches up properties with paths for conversion.
+// This serves to match up properties that were flattened from the original location,
+// even if they've ended up with different names.
+// sourceProperties is a set of endpoints that can be read from.
+// destinationProperties is a set of endpoints that can be written to.
+// assign is a function that will be called for each matching property, with the source and destination endpoints
+// for that property.
+// Returns an error if any of the assignments fail.
+func (builder *PropertyAssignmentFunctionBuilder) selectIdenticallyPathedProperties(
```

Comment:
extremely minor: 
```suggestion
func (builder *PropertyAssignmentFunctionBuilder) selectPropertiesWithIdenticalPaths(
```

(avoids `IdenticallyPathed` which read weird to me. Not sure it's really all that much better.)

Reply:
Done.